packages/shorebird_cli/lib/src/shorebird_ios_release_version_mixin.dart

```diff
@@ -0,0 +1,97 @@
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+import 'package:propertylistserialization/propertylistserialization.dart';
+import 'package:shorebird_cli/src/command.dart';
+
+/// Helpers to determine the release and build number of an iOS app.
+mixin ShorebirdIosReleaseVersionMixin on ShorebirdCommand {
+  /// This key is a user-visible string for the version of the bundle. The
+  /// required format is three period-separated integers, such as 10.14.1. The
+  /// string can only contain numeric characters (0-9) and periods.
+  ///
+  /// See https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring
+  static const releaseVersionKey = 'CFBundleShortVersionString';
+
+  /// The version of the build that identifies an iteration of the bundle.
+  ///
+  /// See https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleversion
+  static const buildNumberKey = 'CFBundleVersion';
+
+  /// Checks the iOS Info.plist and referenced .xcconfig files to determine the
+  /// app's release version and build number.
+  Future<String> determineIosReleaseVersion() async {
```

Comment:
As discussed can we build the ipa/xcarchive and derive the release version from the artifacts similar to how android works?

Reply:
Done.